src/odemis/util/spot.py

```diff
@@ -277,16 +277,22 @@ def MaximaFind(image, qty, len_object=18):
     w = len_object // 2
     refined_center = numpy.zeros_like(pos)
     pos = numpy.rint(pos).astype(numpy.int16)
+    y_max, x_max = image.shape
     for idx, xy in enumerate(pos):
         x_start, y_start = xy - w + 1
         x_end, y_end = xy + w
-        # if spot is close to edge of image x and y can be smaller than 0, crop so the spot is still in the center.
+        # If the spot is near the edge of the image, crop so it is still in the center of the sub-image
         if x_start < 0:
-            x_end = x_end + x_start
+            x_end += x_start
             x_start = 0
+        elif x_end > x_max:
+            x_start += x_end - x_max
```

Comment:
`-=`, as you want `x_start` to be smaller than before, right?
Also, I'd recommend to explicitly write `x_end = x_max`. It's not necessary, as Python will automatically clip the range, but clearer when reading the code.

Reply:
No you want the spot to still be in the center, so x_start should be larger than before.